components/script/dom/htmllabelelement.rs

```diff
@@ -40,6 +42,43 @@ impl HTMLLabelElementMethods for HTMLLabelElement {
     fn GetForm(&self) -> Option<Root<HTMLFormElement>> {
         self.form_owner()
     }
+
+    // https://html.spec.whatwg.org/multipage/#dom-label-htmlfor
+    make_getter!(HtmlFor, "for");
+
+    // https://html.spec.whatwg.org/multipage/#dom-label-htmlfor
+    make_setter!(SetHtmlFor, "for");
+
+    // https://html.spec.whatwg.org/multipage/#dom-label-control
+    fn GetControl(&self) -> Option<Root<HTMLElement>> {
+        let for_ = self.HtmlFor();
+
+        if for_.is_empty() {
```

Comment:
This is supposed to be hasAttribute; https://github.com/servo/servo/blob/0fc9410f10c1122a05cd36bdf8b7faff12c39555/tests/wpt/web-platform-tests/html/semantics/forms/the-label-element/label-attributes.html has a test for this.